[ADD] l10n_tw: new doc (#16659)
odoo/documentation: content/applications/finance/fiscal_localizations/taiwan.rst

======
Taiwan
======

Modules
=======

The following modules related to Taiwan localization are available:

.. list-table::
   :header-rows: 1

   * - Name
     - Technical name
     - Description
   * - :guilabel:`Taiwan - Accounting`
     - `l10n_tw`
     - This is the base module to manage the accounting chart for Taiwan.
   * - :guilabel:`Taiwan - E-invoicing`
     - `l10n_tw_edi_ecpay`
     - This module allows the user to send their invoices to the Ecpay system.
   * - :guilabel:`Taiwan - E-invoicing Ecommerce`
     - `l10n_tw_edi_ecpay_website_sale`
     - This module allows the user to input Ecpay information in ecommerce for sending their invoices to the Ecpay system.
   * - :guilabel:`Taiwan - Accounting Reports`
     - `l10n_tw_reports`
     - This module includes the accounting reports for Taiwan.

> 

Reviewer: We usually add this note on every localization page.
```suggestion

.. note::
   In some cases, such as when upgrading to a version with additional modules, modules may not be
   installed automatically. Any missing modules can be manually :ref:`installed <general/install>`.
   
```

> Accounting

Reviewer: ```suggestion
.. _localizations/taiwan/accounting:

Accounting
```

==========

> ECPay Integration

Reviewer: ```suggestion
ECPay integration
```

-----------------

Odoo supports integration with ECPay to submit generated invoices directly to their system. The

> :guilabel:`Taiwan - E-invoicing` module (`l10n_tw_edi_ecpay`) is required for this functionality.

Reviewer: ```suggestion
.. note::
   Make sure to :ref:`install <general/install>` the :guilabel:`Taiwan - E-invoicing` 
   (`l10n_tw_edi_ecpay`) module.

Odoo supports integration with ECPay to submit generated invoices directly to their system.
```

- :guilabel:`Hashkey`
- :guilabel:`HashIV`

.. image:: taiwan/ECPay-set-up.png
    :alt: ECPay set-up

Configuration in Odoo
*********************

Open the :guilabel:`Accounting` app, navigate to the :guilabel:`Taiwan Electronic Invoicing`
section, and fill in the :guilabel:`MerchantID`, :guilabel:`HashKey`, and :guilabel:`HashIV`
fields.

.. image:: taiwan/ECPay-setting-in-odoo.png
    :alt: ECPay Setting in Odoo

.. note::
   For testing purposes, do not use real credentials. Instead, use the
   `Staging information <https://developers.ecpay.com.tw/?p=24174>`_ provided by ECPay and enable
   the :guilabel:`Staging mode` field.

Contacts
^^^^^^^^

In Taiwan, a clear distinction exists between :abbr:`B2B (Business-to-Business)` and
:abbr:`B2C (Business-to-Consumer)` invoices. This distinction is managed via the
:guilabel:`Contacts` app configuration.

- **Company (B2B)**: When the customer is a company, select the :guilabel:`Company` option and
  ensure the :guilabel:`Tax ID` is filled. The :guilabel:`Email` or :guilabel:`Phone` fields should
  also be filled in.

- **Individual (B2C)**: When the customer is an individual, select the :guilabel:`Individual`

>   option. Provide an :guilabel:`Email` or :guilabel:`Phone` number.

Reviewer: I rephrased it a bit to simplify and avoid duplicating information available in the Contacts page.
For your info, there's no more contact type starting from 19.1.
```suggestion
- **Company (B2B)**: Select the :doc:`contact <../../essentials/contacts>` type, enter the
  :guilabel:`Tax ID`, and provide the :guilabel:`Email` or :guilabel:`Phone` number.
- **Individual (B2C)**: Select the :doc:`contact <../../essentials/contacts>` type and provide the
  :guilabel:`Email` or :guilabel:`Phone` number.
```

     :guilabel:`Carrier Number` (and Carrier Number 2 if required) when prompted.

#. :guilabel:`Confirm` the invoice, then click :guilabel:`Print & Send`. Ensure the
   :guilabel:`Send to Ecpay` checkbox is selected.
#. The :guilabel:`Invoice Status`, :guilabel:`Ecpay Invoice Number`, and :guilabel:`Creation Date`
   in the :guilabel:`Ecpay` tab will update automatically upon successful submission.

   .. image:: taiwan/ECPay-tab.png
       :alt: ECPay tab

.. note::
   ECpay invoice is also accessible in Odoo when the :guilabel:`Get Printed Version` option is
   checked:
   click the :icon:`fa-cog` (gear) icon, then select :guilabel:`Print Ecpay invoice`. Select the
   :guilabel:`Print Format (B2C)` field, and then click the :guilabel:`Print Invoice` button.

B2B Invoice (Company)
^^^^^^^^^^^^^^^^^^^^^

#. Create an invoice and select a :guilabel:`Company` customer (or an individual belonging to a
   company).
#. Confirm the invoice, click :guilabel:`Print & Send`, and ensure :guilabel:`Send to Ecpay` is
   selected.
#. The :guilabel:`Invoice Status`, :guilabel:`Ecpay Invoice Number`, and :guilabel:`Creation Date`

>    in the :guilabel:`Ecpay` tab will update automatically upon successful submission.

Reviewer: ```suggestion
To send a B2B invoice, follow these steps:

#. Navigate to :menuselection:`Accounting --> Customers --> Invoices` and :ref:`create a new
   invoice <accounting/invoice/creation>`.
#. Select a :ref:`customer <localizations/taiwan/contacts>` who is either a :guilabel:`Company` or 
   an :guilabel:`Individual` belonging to a :guilabel:`Company`.
#. :guilabel:`Confirm` the invoice, and click :guilabel:`Send`. Then, ensure the
   :guilabel:`Send to Ecpay` checkbox is selected, and click :guilabel:`Send`.
```

.. important::
   Ensure the :guilabel:`Tax ID` field of the customer is filled in with the Tax Identification
   Number; otherwise the invoice will not be sent to ECPay successfully.

.. note::
   - For the B2B invoice, the Ecpay invoice can be printed out/ downloaded by default: click the
     :icon:`fa-cog` (gear) icon, then click :guilabel:`Print Ecpay invoice`.
   - The user can also check the :guilabel:`Ecpay invoice number` and the
     :guilabel:`Invoice status` in the :guilabel:`log note`.

Invoice Cancellation
********************

To cancel a submitted invoice:

#. Open the invoice and click :guilabel:`Request Cancel`.
#. In the pop-up window, provide a cancellation :guilabel:`Reason`, then click
   :guilabel:`Cancel Invoice`.
#. The :guilabel:`Invoice Status` will change to :guilabel:`Invalid`, and the

>    :guilabel:`Invalidate Reason` will be recorded in the :guilabel:`Ecpay` tab.

Reviewer: ```suggestion
#. The :guilabel:`Invoice Status` changes to :guilabel:`Invalid`, and the :guilabel:`Invalidate 
   Reason` is filled in with the cancellation :guilabel:`Reason`.
```

   .. image:: taiwan/ECPay-invoice-cancellation.png
       :alt: ECPay invoice cancellation

Send credit notes to ECPay
**************************

B2C Credit Notes
^^^^^^^^^^^^^^^^

Before sending a credit note, the original invoice must be successfully submitted to ECPay. To send

> a credit note to ECPay,

Reviewer: ```suggestion
Before sending a credit note, the original invoice must be successfully submitted to ECPay. 

B2C Credit Notes
^^^^^^^^^^^^^^^^

To send a B2C credit note to ECPay, follow these steps:
```

#. Click :guilabel:`Credit Note` on the invoice.
#. Enter the :guilabel:`Reason displayed on Credit Note`.
#. Select the :guilabel:`Agreement Type`.
#. Select the :guilabel:`Allowance Notify Way`.
#. Click :guilabel:`Confirm`, then :guilabel:`Print & Send`, ensuring
   :guilabel:`Send to Ecpay (Issue Allowance)` is selected.

The :guilabel:`Refund State` will update to :guilabel:`Agreed` (for Offline agreements) or
:guilabel:`To be agreed` (for Online agreements); the :guilabel:`Refund Invoice Number`,
:guilabel:`Refund invoice Agreement Type` and :guilabel:`Allowance Notify Way` will also be updated
accordingly.

B2B credit notes
^^^^^^^^^^^^^^^^

#. Click :guilabel:`Credit Note` and enter the :guilabel:`Reason displayed on Credit Note`.
#. Click :guilabel:`Confirm`, then :guilabel:`Print & Send`, ensuring
   :guilabel:`Send to Ecpay (Issue Allowance)` is selected.

> #. The :guilabel:`Refund Invoice Number` is updated automatically.

Reviewer: When I tested it for a B2B invoice, the Agreement Type and Allowance Notify Way fields were available. If so, the following step should be added: Select the Agreement Type and the Allowance Notify Way, and click Reverse.
```suggestion
To send a B2B credit note to ECPay, follow these steps:

#. Click :guilabel:`Credit Note` on the relevant invoice and enter the :guilabel:`Reason displayed
   on Credit Note`.
#. Click :guilabel:`Confirm`, then :guilabel:`Send`, ensuring :guilabel:`Send to Ecpay (Issue 
   Allowance)` is selected.
#. The :guilabel:`Refund Invoice Number` is updated automatically.
```

This ensures invoices are generated and sent to ECPay automatically once online payment is confirmed.

.. image:: taiwan/ECPay-ecommerce-auto-invoice.png
   :alt: ECPay Ecommerce auto invoice

Checkout Process
----------------

- **B2C Customers**: During checkout, customers must select :guilabel:`Request a paper copy` to
  receive a physical invoice. If :guilabel:`No` is selected, no copy is provided. If
  :guilabel:`Yes`, the invoice can be downloaded from the invoice view.

  .. image:: taiwan/ecommerce-b2c-checkout.png
     :alt: Ecommerce B2C Checkout

- **B2B Customers**: An :guilabel:`Invoicing Info` page appears after delivery confirmation.
  Customers can choose to donate the invoice or select an :guilabel:`Ecpay e-invoice carrier`.

  .. image:: taiwan/ecommerce-b2b-checkout.png
     :alt: Ecommerce B2B Checkout

  .. note::
   - If no option has been selected, a Print Ecpay invoice can still be downloaded from the
     invoice view.
   - If an invoice fails to be sent to ECPay, error messages can be reviewed in the
     :guilabel:`Log Note` of the invoice. Successful transmissions are verified in the

>      :guilabel:`Ecpay` tab.

Author: Hello @afma-odoo , agree! I moved this part to the note of accounting.

- :ref:`E-invoicing (ECPay) <localizations/taiwan/ecpay-integration>`: integration for electronic
  invoicing in line with Taiwanese government requirements.

.. _localizations/taiwan/accounting:

Accounting
==========

.. _localizations/taiwan/ecpay-integration:

ECPay integration
-----------------

.. note::
   Make sure to :ref:`install <general/install>` the :guilabel:`Taiwan - E-invoicing`
   (`l10n_tw_edi_ecpay`) module.

Odoo supports integration with ECPay to submit generated invoices directly to their system.

.. _localizations/taiwan/ecpay-setup:

Set-up
~~~~~~

To configure the integration, valid credentials must be retrieved from the
`ECPay vendor backend <https://vendor.ecpay.com.tw/User/LogOn_Step1>`_. Go to
:menuselection:`System Settings --> System Integration Settings --> Integration Information` to get
the required information:

- :guilabel:`MerchantID`
- :guilabel:`Hashkey`
- :guilabel:`HashIV`

.. _localizations/taiwan/odoo-configuration:

Configuration in Odoo
*********************

Go to :menuselection:`Accounting --> Configuration --> Settings`, scroll down to the
:guilabel:`Taiwan Electronic Invoicing` section, and fill in the :guilabel:`MerchantID`,

> :guilabel:`HashKey`, and :guilabel:`HashIV` fields.

Reviewer: ```suggestion
:guilabel:`HashKey`, and :guilabel:`HashIV` fields, retrieved from the `ECPay vendor backend 
<localizations/taiwan/ecpay-setup>`.
```

   :guilabel:`Reason displayed on Credit Note`.
#. Select the :guilabel:`Agreement Type` and the :guilabel:`Allowance Notify Way`, and click
   :guilabel:`Reverse`.
#. Click :guilabel:`Confirm`, then :guilabel:`Send`, ensuring
   :guilabel:`Send to Ecpay (Issue Allowance)` is selected.

Then, The :guilabel:`Refund State` updates to :guilabel:`Agreed`
(for :guilabel:`Offline agreements`) or :guilabel:`To be agreed`
(for :guilabel:`Online agreements`). Additionally, the :guilabel:`Refund Invoice Number`,
:guilabel:`Refund invoice Agreement Type` and :guilabel:`Allowance Notify Way` updates accordingly
in the :guilabel:`Ecpay` tab.

.. _localizations/taiwan/B2B-credit-notes:

B2B credit notes
^^^^^^^^^^^^^^^^

To send a B2B credit note to ECPay, follow these steps:

#. Click :guilabel:`Credit Note` on the relevant invoice, enter the
   :guilabel:`Reason displayed on Credit Note`, then click :guilabel:`Reverse`.
#. Click :guilabel:`Confirm`, then :guilabel:`Send`, ensuring
   :guilabel:`Send to Ecpay (Issue Allowance)` is selected.

Then the :guilabel:`Refund Invoice Number` is updated automatically in the :guilabel:`Ecpay` tab.

.. note::

>    The credit note amount cannot exceed the original invoice amount.

Author: Moved it to the "Send credit notes to ECPay" section
